wordpress-mobile/WordPress-Android
# My Site Dashboard - Implement quick link ribbons

Fixes #16005

This PR adds the changes for implementing the Quick Link Ribbon in Dashboard(Home) tab. Quick link Ribbon should behave the same way as the quick link actions on the Site menu tab. The navigation from the Quick Link Ribbon should behave in the same way as Quick Actions on the Site Menu Tab. 

| | |  
|---|---|
| ![Screenshot_20220405_002932](https://user-images.githubusercontent.com/17463767/161613270-d57201db-7437-4e48-ad6b-67ce950dc4b3.png)  |  ![Screenshot_20220405_002852](https://user-images.githubusercontent.com/17463767/161613282-f91d3fba-8595-4493-8115-89d083646d22.png) |  


The above UI - Quick Link Ribbon should have the same behaviour as Quick Link Action Card in the site menu, which is shown below. 

<img src="https://user-images.githubusercontent.com/17463767/161615111-4bfcc454-b0ea-424d-a0d5-4389edd9fbc2.png" width="370" height="680">
 

To test:
- Go to My Site -> Me -> App Settings -> Test Feature Configuration.
- Turn the MySiteDashboardPhase2FeatureConfig feature flag ON.
- Relaunch the app.
- Login with an account having wp com sites and go to the My Site tab.
- Notice that in the Home tab, Quick link ribbons are shown 
- Notice that the order of the Quick link ribbons are Pages, Posts, Media and Stats
- Clicking on each quick link ribbon item should take the user to the same destination as the one in quick link action card. 


### Known Issues 
####  [Quick start focus point on quick link ribbon](https://github.com/wordpress-mobile/WordPress-Android/issues/16227)

## Regression Notes
1. Potential unintended areas of impact
- Quick Link action card not shown 
- Quick Link Ribbon does not navigate the User to the correct destination 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and Unit Test

3. What automated tests I added (or what prevented me from doing so)
Manual tests

PR submission checklist:
